### ISSUE TITLE
azurerm_key_vault_certificate does not support dns names or SAN

### ISSUE THREAD

**Issue by LaurentLesle** [terraform-providers/terraform-provider-azurerm#1808]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When creating a certificate request for KeyVault it is not possible to specify an array of DNS names (or Subject Alternate Names - SAN) to the request.

This is something supported by the PowerShell command New-AzureKeyVaultCertificatePolicy -DnsName - [link here](https://docs.microsoft.com/en-us/powershell/module/azurerm.keyvault/new-azurekeyvaultcertificatepolicy?view=azurermps-6.7.0)

### New or Affected Resource(s)

* azurerm_key_vault_certificate - attribute certificate_policy.x509_certificate_properties

### Potential Terraform Configuration

```hcl
x509_certificate_properties {
      key_usage = [
        "cRLSign",
        "dataEncipherment",
        "digitalSignature",
        "keyAgreement",
        "keyCertSign",
        "keyEncipherment",
      ]
      subject = "CN=subjectName"

      # proposed implementation:
       dns_name = [
           "*.test.local",
           "*.test.internal"
       ]
      validity_in_months = 12
    }
```

### References

* #0000

**Comment by draggeta** [comment]
@LaurentLesle this has been added in 1.18

**Comment by tombuildsstuff** [comment]
Closing as support for this has landed in v1.18 - you can upgrade to this by specifying this in the Provider Block (as shown below) and then running `terraform init -upgrade`:

```
provider "azurerm" {
  version= "=1.18.0"
}
```

Thanks!
